### PR TITLE
cli --min-packsize and --file-read-concurrency

### PR DESCRIPTION
**Pull request by madsi1m**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Would like to add the following in order to tune restic rather than hardcoding it.
--min-packsize
--file-read-concurrency

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Nope

Checklist
---------

- [X] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [X] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [X] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

**Comment by dimejo** [comment]
While I understand the need to tune such options I think they require proper documentation. People tend to tune such options to the extreme which might cause bottlenecks or performance issues and result in a lot of questions and complains on the forum. Hence the documentation should cover these questions:

- When is it necessary to tune these options?
- What are sane maximum values?
- What happens if these options are set too high (e.g. `--file-read-concurrency 100`)?
- Is it possible to change `--min-packsize` between 2 backup runs? What are the impacts?

**Comment by MichaelEischer** [comment]
Does `--file-read-concurrency` have to be a global-option? Right now it's exclusively used by the backup command. `--min-packsize` is also not used by that many commands, actually just by `backup`, `prune` and `forget --prune`. But I guess that makes the option simliar to `--limit-upload`.

**Comment by madsi1m** [comment]
--file-read-concurrency doesn't need to be a global option, you are quite right it is only used by the backup command.

File: changelog/unreleased/pull-2671 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Feature: Addition of --min-packsize and --file-read-concurrency
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-2671:1]: Judging from the changelog template https://github.com/restic/restic/blob/master/changelog/TEMPLATE this should read `Add [...] flags`

File: changelog/unreleased/pull-2671 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+In order to tune restic for large installs we need to be able to configure --min-packsize and --file-read-concurrency. In version 0.9.6 these numbers are hard coded, this pull request allows you to adjust these.
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-2671:3]: The changelog usually doesn't refer explicitly to previous versions. The description should also be written in past tense. There's also no need to mention that this was change via a pull request. (Sorry for the nitpicking)

File: cmd/restic/global.go [hunk 3]
```diff
@@ -109,8 +128,18 @@ func init() {
 	f.BoolVar(&globalOptions.CleanupCache, "cleanup-cache", false, "auto remove old cache directories")
 	f.IntVar(&globalOptions.LimitUploadKb, "limit-upload", 0, "limits uploads to a maximum rate in KiB/s. (default: unlimited)")
 	f.IntVar(&globalOptions.LimitDownloadKb, "limit-download", 0, "limits downloads to a maximum rate in KiB/s. (default: unlimited)")
+	f.UintVar(&globalOptions.MinPackSize, "min-packsize", 0, "set min pack size in MiB. (default: $RESTIC_MIN_PACKSIZE or 4)")
```
Comment (MichaelEischer) [inline cmd/restic/global.go:131]: If you put backticks '\`'  around the word `size` then cobra will use that to name the flag parameter.

**Comment by metalsp0rk** [comment]
@madsi1m I'd love to see this included.  Do you have the bandwidth to finish this up, or would you mind if I took over?

**Comment by madsi1m** [comment]
@metalsp0rk sadly I don't have the bandwidth at the moment. Go for it. Would be great to have this in master 👍

**Comment by metalsp0rk** [comment]
No worries, I just didn't want to step on any toes.  :+1:

**Comment by metalsp0rk** [comment]
I've submitted a new PR for this, #2750

**Comment by madsi1m** [comment]
cool

**Comment by aawsome** [comment]
What about closing this now that #2750 is almost finished?

**Comment by MichaelEischer** [comment]
Closing this in favor of #2750.

**Pull request closed without merging**